src/frontend/src/optimizer/plan_node/logical_join.rs

```diff
@@ -1054,22 +1054,22 @@ impl LogicalJoin {
         let lookup_prefix_len = reorder_idx.len();
         let predicate = predicate.reorder(&reorder_idx);
 
-        let left = if dist_key_in_order_key_pos.is_empty() {
-            self.left()
-                .to_stream_with_dist_required(&RequiredDist::single(), ctx)?
+        let required_dist = if dist_key_in_order_key_pos.is_empty() {
+            RequiredDist::single()
         } else {
             let left_eq_indexes = predicate.left_eq_indexes();
             let left_dist_key = dist_key_in_order_key_pos
                 .iter()
                 .map(|pos| left_eq_indexes[*pos])
                 .collect_vec();
 
-            self.left().to_stream_with_dist_required(
-                &RequiredDist::shard_by_key(self.left().schema().len(), &left_dist_key),
-                ctx,
-            )?
+            RequiredDist::shard_by_key(self.left().schema().len(), &left_dist_key)
```

Comment:
Not sure if it is ok. why is it not be `RequiredDist::PhysicalDist(Distribution::HashShard(left_dist_key))`

Reply:
Oh, I get it, because it is `enforce` instead of `enforce_if_not_satisfy`...

Reply:
Use `RequiredDist::hash_shard` instead.